src/session.rs

```diff
@@ -0,0 +1,83 @@
+extern crate dirs;
+
+use crate::auth::Auth;
+use crate::request_items::Parameter;
+use crate::utils::ensure_session_dir_exists;
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::io::prelude::*;
+
+const VERSION: &'static str = env!("CARGO_PKG_VERSION");
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Session {
+    pub identifier: String,
+    pub host: String,
+    pub headers: Vec<Parameter>,
+    pub auth: Option<Auth>,
+    ht_version: String,
+}
+
+impl Session {
+    pub fn new(identifier: String, host: &str, headers: Vec<Parameter>, auth: Option<Auth>) -> Session {
+        Session {
+            identifier,
+            host: host.to_string(),
+            headers,
+            auth,
+            ht_version: String::from(VERSION),
+        }
+    }
+
+    pub fn save(&self) -> std::io::Result<String> {
```

Comment:
Documenting what value this returns (the session's identifier) would be helpful.
Currently the function signature tells us _some `String`_ but without docs that may as well be a hardcoded `"test".into()` given what I know as a user.

Reply:
Ok, I handle this below, it's no longer needed I think.